### PR TITLE
Octo fixes

### PR DESCRIPTION
**Pull request by bgrozev**

### CODE REVIEW

File: src/main/java/org/jitsi/jicofo/FocusManager.java [hunk 1]
```diff
@@ -485,7 +485,8 @@ private String generateConferenceId()
             {
                 id
                     = jicofoShortId +
-                        Integer.toHexString(RANDOM.nextInt(0x1_0000));
+                        String.format(
+                            "%4h", RANDOM.nextInt(0x1_0000)).replace(' ', '0');
```
Comment (bbaldino) [inline src/main/java/org/jitsi/jicofo/FocusManager.java]: what was the issue here?

File: src/main/java/org/jitsi/jicofo/JitsiMeetConferenceImpl.java [hunk 2]
```diff
@@ -1433,12 +1434,34 @@ public XMPPError onSessionAccept(
         participant.setRTPDescription(answer);
         participant.addTransportFromJingle(answer);
 
+        MediaSourceMap sourcesAdvertised
+                = MediaSourceMap.getSourcesFromContent(answer);
+        MediaSourceGroupMap sourceGroupsAdvertised
+                = MediaSourceGroupMap.getSourceGroupsForContents(answer);
+        if (sourcesAdvertised.isEmpty())
```
Comment (bbaldino) [inline src/main/java/org/jitsi/jicofo/JitsiMeetConferenceImpl.java]: i know we talked about this, but there is a reason we *can't* just do this on the client, right?  i worry we'll introduce new issues by injecting fake ssrcs that the client isn't aware of.

Comment (bgrozev) [reply src/main/java/org/jitsi/jicofo/JitsiMeetConferenceImpl.java]: I don't know, I haven't looked at the client at all. I agree with your concern, but I think it also applies to some extent if we make client changes.

File: src/main/java/org/jitsi/jicofo/FocusManager.java (hunk 1 above)
Comment (bgrozev) [reply src/main/java/org/jitsi/jicofo/FocusManager.java]: The conference ID consists of one byte for an ID of the jicofo instance plus two bytes for the conference within jicofo. If the random number was < 0x1000 we'd get an ID like "ff123" (with jicofo ID "ff"). This gets serialized the same as 0ff123 which could clash with jicofo ID "0f" and conf ID "f123".

This change pads the random ID so for the case above we get "ff0123".

File: src/main/java/org/jitsi/jicofo/JitsiMeetConferenceImpl.java (hunk 2 above)
Comment (bbaldino) [reply src/main/java/org/jitsi/jicofo/JitsiMeetConferenceImpl.java]: the client would be a single source of truth with regards to its sources though.

File: src/main/java/org/jitsi/jicofo/FocusManager.java (hunk 1 above)
Comment (bbaldino) [reply src/main/java/org/jitsi/jicofo/FocusManager.java]: it looks like `String.format("%04x", ...)` does what you need and i think is cleaner than the replace:

```
String.format("%04x", 5); // --> 0005
String.format("%04x", 0xFFFF); // --> ffff
```

File: src/main/java/org/jitsi/jicofo/JitsiMeetConferenceImpl.java (hunk 2 above)
Comment (gpolitis) [reply src/main/java/org/jitsi/jicofo/JitsiMeetConferenceImpl.java]: > the client would be a single source of truth with regards to its sources though.

True, at least temporarily, until ssrcs fade away completely.

**Comment by bbaldino** [comment]
I don't follow, it would be the client who removes them, right? So it'd
still be the single source of truth.

**Comment by bgrozev** [comment]
Okay, I did the digging. The client intentionally does NOT generate an SSRC when it's receive only. It was disabled [here](https://github.com/jitsi/lib-jitsi-meet/pull/466) as a workaround for (this chrome bug)[https://bugs.chromium.org/p/webrtc/issues/detail?id=7555]. The Chrome bug is still not fixed, and simulcast fails to be enabled if we generate the SSRC early. 

So, I'll go ahead with this "fix".

**Pull request merged**